packages/airplane/internal/params/params_serialized.ts

```diff
@@ -61,7 +68,106 @@ export const serializeParamValue = (paramValue: InputParamValue): SerializedPara
     return paramValue.id;
   }
   if (paramValue instanceof Date) {
-    return toISOString(paramValue);
+    // TODO: Currently we incorrectly serialize `date` parameters the same as `datetime`.
+    //       `date` parameters need to instead be formatted as `YYYY-MM-DD`.
+    return paramValue.toISOString();
   }
+  // TODO: For string date/datetimes, consider parsing the strings as Date's and reserializing
+  //       to the correct format.
   return paramValue;
 };
+
+const serializeParamType = (
+  t: ParamType
+): [SerializedParamType, SerializedParamComponent | undefined] => {
+  switch (t) {
+    case "shorttext":
+      return ["string", undefined];
+    case "longtext":
+      return ["string", "textarea"];
+    case "sql":
+      return ["string", "editor-sql"];
+    case "file":
+      return ["upload", undefined];
+    default:
+      return [t, undefined];
+  }
+};
+
+export const serializeParams = (params: InputParams): SerializedParams => {
+  const serializedParams = [];
+  for (const slug in params) {
+    const param = params[slug];
+    if (typeof param === "string") {
+      const [type, component] = serializeParamType(param);
+      serializedParams.push({
+        slug: slug,
+        name: slug,
+        type: type,
+        component: component,
+        constraints: {
+          optional: false,
+        },
+      });
+    } else {
+      const [type, component] = serializeParamType(param.type);
+      const regex = "regex" in param ? param.regex : undefined;
+      const optional = param.required === false ? true : false;
+      serializedParams.push({
+        slug: slug,
+        name: param.name ? param.name : slug,
+        type: type,
+        component: component,
+        desc: param.description && dedent(param.description),
+        default: serializeParamValue(param.default),
+        constraints: {
+          optional: optional,
+          options:
+            "options" in param && param.options != null
+              ? // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                serializeOptions(param.options as any, slug, optional)
+              : undefined,
+          regex,
+        },
+      });
+    }
+  }
+  return serializedParams;
+};
+
+const serializeOptions = <TOption extends InputParamValueByType[keyof InputParamValueByType]>(
+  options: Array<TOption | SelectOption<TOption>>,
+  slug: string,
+  optional: boolean
+): Array<{ label: string; value: SerializedParamValue }> => {
+  if (options.length === 0 && !optional) {
+    throw new Error(`The required prompt parameter, "${slug}", has no options to select from.`);
```

Comment:
This function isn't directly related to prompts.

```suggestion
    throw new Error(`The required parameter, "${slug}", has no options to select from.`);
```

Reply:
Hmm, does this match our API semantics? I would almost assume if a user passes an empty array of options, we would ignore it.

Reply:
yup you're right. should we add the check to parameter verification?

<img width="484" alt="Screen Shot 2022-11-28 at 3 40 56 PM" src="https://user-images.githubusercontent.com/11494140/204376548-b248f6d3-3264-4d33-a7d4-7661fb7d744c.png">

Reply:
We apparently do this in our UI:

<img width="784" alt="CleanShot 2022-11-28 at 18 05 42" src="https://user-images.githubusercontent.com/2907397/204399442-f0583f58-8aaf-4126-983b-bbe340ee675d.png">

Returning an error from the API SGTM. Can you add a Linear for that?

Reply:
https://linear.app/airplane/issue/TSK-1193/add-parameter-verification-check-that-empty-options-is-not-allowed